babel/index.js

```diff
@@ -36,7 +34,11 @@ var BabelClient = function babelClient(config) {
         throw new Error('Invalid Babel config: babel_host');
     }
 
-    this.config.babel_hostname = this.config.babel_host.split('://', 2)[1];
+    var port = config.babel_port ? config.babel_port : (config.babel_host === 'https:' ? 443 : 80);
+    this.config.endpointUrl = new URL(`${config.babel_host}:${port}`);
+
+    var schema = this.config.endpointUrl.protocol;
+    this.http = require(schema === 'https:' ? 'https' : 'http');
```

Comment:
nitpick: it's [scheme](https://datatracker.ietf.org/doc/html/rfc1738#section-2.1)
```suggestion
    var scheme = this.config.endpointUrl.protocol;
    this.http = require(scheme === 'https:' ? 'https' : 'http');
```